File: backend/app/schemas/academy.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class LessonSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    title: str
    order_index: int
    duration_minutes: int
    visual_type: str
    status: str | None = None


class CourseSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    title: str
    description: str
    category: str
    difficulty: str
    estimated_hours: float
    lesson_count: int = 0
    progress_percent: float | None = None
    is_enrolled: bool = False


class CourseDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    title: str
    description: str
    category: str
    difficulty: str
    estimated_hours: float
    disclaimer: str | None = None
    lessons: list[LessonSummary] = []
    progress_percent: float | None = None
    is_enrolled: bool = False


class LessonDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    title: str
    content_type: str
    content: str
    visual_type: str
    visual_data: dict | None = None
    quiz: dict | None = None
    duration_minutes: int
    course_slug: str
    course_title: str
    status: str | None = None
    quiz_score: float | None = None


class EnrollmentResponse(BaseModel):
    course_slug: str
    progress_percent: float
    enrolled_at: datetime


class LessonProgressUpdate(BaseModel):
    status: str = Field(..., pattern="^(in_progress|completed)$")
    quiz_score: float | None = Field(default=None, ge=0, le=100)
    time_spent_seconds: int | None = Field(default=None, ge=0)


class UserLearningSummary(BaseModel):
    enrolled_courses: int
    completed_lessons: int
    total_lessons: int
    overall_progress_percent: float
    courses: list[CourseSummary]

File: frontend/src/types/academy.ts
export interface LessonSummary {
  id: number
  slug: string
  title: string
  order_index: number
  duration_minutes: number
  visual_type: string
  status?: string | null
}

export interface CourseSummary {
  id: number
  slug: string
  title: string
  description: string
  category: string
  difficulty: string
  estimated_hours: number
  lesson_count: number
  progress_percent?: number | null
  is_enrolled: boolean
}

export interface CourseDetail extends Omit<CourseSummary, 'lesson_count'> {
  disclaimer?: string | null
  lessons: LessonSummary[]
}

export interface LessonDetail {
  id: number
  slug: string
  title: string
  content_type: string
  content: string
  visual_type: string
  visual_data?: Record<string, unknown> | null
  quiz?: QuizData | null
  duration_minutes: number
  course_slug: string
  course_title: string
  status?: string | null
  quiz_score?: number | null
}

export interface QuizQuestion {
  id: string
  question: string
  options: string[]
  correct_index: number
  explanation: string
}

export interface QuizData {
  questions: QuizQuestion[]
  passing_score: number
}

export interface UserLearningSummary {
  enrolled_courses: number
  completed_lessons: number
  total_lessons: number
  overall_progress_percent: number
  courses: CourseSummary[]
}

export const CATEGORY_LABELS: Record<string, string> = {
  rf: 'RF',
  sdr: 'SDR',
  hackrf: 'HackRF',
  signals: 'Señales',
  hardware: 'Hardware',
}

export const DIFFICULTY_LABELS: Record<string, string> = {
  beginner: 'Principiante',
  intermediate: 'Intermedio',
  advanced: 'Avanzado',
}
